File: core/src/test/scala/diplomacy/board/NameSpec.scala
package diplomacy.board

import diplomacy.UnitSpec

class NameSpec extends UnitSpec {
  "A Name" should {
    "be stringify using its abbriviated name." in {
      val eng = Name("English Channel", "Eng")
      eng.toString should be("Eng")
    }
  }
  it when {
    "created" can {
      "omit an abbriviated name." in {
        val swe = Name("Sweden")
        swe.toString should be("Sweden")
      }
    }
  }
}

File: test/board/name-spec.ts
import * as chai from "chai"
import { Name } from "./../../src/board/name"

chai.should()

describe("Name", () => {
  it("receives a name and an abbreviated name.", () => {
    const name = new Name("English Channel", "Eng")

    name.name.should.equal("English Channel")
    name.abbreviatedName.should.equal("Eng")
  })
  it("omit an abbreviated name", () => {
    const name = new Name("Sweden")

    name.abbreviatedName.should.equal("Sweden")
  })
  it("implements a custom toString.", () => {
    const name = new Name("English Channel", "Eng")

    name.toString().should.equal("Eng")
  })
})
